advanced_source/extend_dispatcher.rst

```diff
@@ -0,0 +1,350 @@
+  Tensor & abs_out(Tensor & out, const Tensor & self); // {"schema": "aten::abs.out(Tensor self, *, Tensor(a!) out) -> Tensor(a!)", "dispatch": "True", "default": "False"}
+  Tensor absolute(const Tensor & self); // {"schema": "aten::absolute(Tensor self) -> Tensor", "dispatch": "False", "default": "False"}
+  Tensor & absolute_(Tensor & self); // {"schema": "aten::absolute_(Tensor(a!) self) -> Tensor(a!)", "dispatch": "False", "default": "False"}
+  Tensor & absolute_out(Tensor & out, const Tensor & self); // {"schema": "aten::absolute.out(Tensor self, *, Tensor(a!) out) -> Tensor(a!)", "dispatch": "False", "default": "False"}
+  Tensor angle(const Tensor & self); // {"schema": "aten::angle(Tensor self) -> Tensor", "dispatch": "True", "default": "True"}
+  Tensor & angle_out(Tensor & out, const Tensor & self); // {"schema": "aten::angle.out(Tensor self, *, Tensor(a!) out) -> Tensor(a!)", "dispatch": "True", "default": "False"}
+  Tensor sgn(const Tensor & self); // {"schema": "aten::sgn(Tensor self) -> Tensor", "dispatch": "True", "default": "True"}
+
+There're multiple fields associated with a single operator. Let's break it down using ``abs_out`` as an example:
+
+* ``Tensor & abs_out(Tensor & out, const Tensor & self);`` is the C++ signature of the operator, your C++
+  kernel should match this signature exactly.
+* ``aten::abs.out(Tensor self, *, Tensor(a!) out) -> Tensor(a!)`` is the unique JIT schema representing the operator.
+  Since C++ allows overloads but JIT schema is guaranteed to be unique, operator name in JIT schema might be different
+  from the C++ signature to differentiate between multiple overloads of the same operator.
+* ``dispatch`` and ``default`` are boolean fields that provide information about what native PyTorch kernels
+  can do, thus implies whether it's required for backend extenders to implement the kernel.
+  More details can be found in :ref:`register kernels for the new backend<register-kernel>`.
+
+
+.. _register-kernel:
+
+Register kernels for the new backend
+------------------------------------
+
+To register your kernels to PyTorch dispatcher, you can use the
+``TORCH_LIBRARY_IMPL`` API described in
+`Registering a Dispatched Operator in C++ <dispatcher>`_:
+
+.. code-block:: cpp
+
+  TORCH_LIBRARY_IMPL(aten, PrivateUse1, m) {
+    m.impl(<schema_my_op1>, &my_op1);
+    m.impl(<schema_my_op2>, &my_op2);
+    m.impl(<schema_my_op2_backward>, &my_op2_backward);
+  }
+
+Now let's zoom in and what operator requires a kernel from a customized backend and what's
+inside the kernels exactly.
+
+PyTorch currently has more than 1600 operators and it’s still growing.  It’s unrealistic
+for backend extensions to keep up with this speed.  Even for native backends like CPU
+or CUDA, it often requires a lot of work to write dedicated kernels for every new op.
+
+Fortunately, the PyTorch dispatcher allows registration to *alias dispatch keys* which can
```

Comment:
nit: as a backend implementor, I don't really care about alias dispatch keys. The second sentence is what I care about: some kernels in PyTorch don't need to be implemented, we've already decomposed them into several known operators, so you only need to implement the known operator.